src/legacy/smartweave-global.ts

```diff
@@ -243,31 +248,35 @@ class Vrf {
 }
 
 export class KV {
-  private _kvBatch: BatchDBOp[] = [];
+  private _kvBatch: BatchDBOp<any>[] = [];
 
-  constructor(private readonly _storage: KVDatabase | null) {}
+  constructor(private readonly _storage: SortKeyCache<any> | null, private readonly _transaction: Transaction) {}
 
-  async put(key: string, value: string): Promise<void> {
+  async put(key: string, value: any): Promise<void> {
     this.checkStorageAvailable();
     this._kvBatch.push({
       type: 'put',
-      key: Buffer.from(key),
-      value: Buffer.from(value)
+      key: new CacheKey(key, this._transaction.sortKey),
+      value: value
     });
   }
 
-  del(key: string): void {
+  async get<V>(key: string): Promise<V | null> {
     this.checkStorageAvailable();
-    this._kvBatch.push({
-      type: 'del',
-      key: Buffer.from(key)
-    });
-  }
+    const sortKey = this._transaction.sortKey;
+
+    if (this._kvBatch.length > 0) {
+      const putBatches = this._kvBatch.filter((batchOp) => batchOp.type === 'put') as PutBatch<any>[];
```

Comment:
we should probably take 'del' operations into consideration here.
Ie. reverse and if the first operation for the given key is del - return null immediately.

Reply:
We cannot just put nulls, and removing the whole key history is also not a option.
Let's consider adding some flag like: `removed` ?

Reply:
del will be - maybe - added in a separate PR.